# Profitable App Profiles for the App Store and Google Play Markets

The aim of this project is to find mobile app profiles that are profitable for the App Store and Google Play markets. We are working as data analysts for a company that builds Android and iOS mobile apps, and our job is to enable our team of developers to make data-driven decisions with respect to the kind of apps they build.

At our company, we only build apps that are free to download and install, and our main source of revenue consists of in-app ads. This means that our revenue for any given app is mostly influenced by the number of users that use our app. Our goal for this project is to analyze data to help our developers understand what kinds of apps are likely to attract more users

### Open and Explore the data
- A data set containing data about approximately ten thousand Android apps from Google Play. You can download the data set directly from [this link ](https://dq-content.s3.amazonaws.com/350/googleplaystore.csv)
- A data set containing data about approximately seven thousand iOS apps from the App Store. You can download the data set directly from [this link](https://dq-content.s3.amazonaws.com/350/AppleStore.csv)


In [61]:
# Read the Android data file 
from csv import reader
with open('googleplaystore.csv',encoding= 'UTF-8') as f :
    lines =  list(reader(f))
android_head = lines[0]
android_data = lines[1:]
#First 5 lines
print("Length of Android data",len(android_data))
print(android_data[:5])

Length of Android data 10841
[['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up'], ['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up'], ['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up'], ['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up'], ['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']]


In [62]:
# Read the Appstore data file 
with open('Applestore.csv',encoding= 'UTF-8') as f :
    lines =  list(reader(f))
ios_head = lines[0]
ios_data = lines[1:]
print("Length of ios data",len(ios_data))
#First 5 lines
print(ios_data[:5])

Length of ios data 7197
[['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1'], ['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1'], ['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1'], ['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1'], ['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']]



# Deleting wrong data

The Google Play data set has a dedicated discussion section, and we can see that [one of the discussions](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015) outlines an error for row 10472. Let's print this row and compare it against the header and another row that is correc



In [63]:
#Delete the wrong data in Android data file 
print("Length of android data is ",len(android_data))
print("Android Header is ", android_head)
print("\n")
print(android_data[1],"\n length of 1st row is ",len(android_data[1]))
print("\n")
print(android_data[10472],"\n length of 10472  row is ",len(android_data[10472])) 
# Row 10472 has missing fields so remove the data
del android_data[10472]
print("Length of android data is ",len(android_data))

Length of android data is  10841
Android Header is  ['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up'] 
 length of 1st row is  13


['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up'] 
 length of 10472  row is  12
Length of android data is  10840


# Removing duplciates Entries

Analyze the data for duplicates entries in each app. If there are duplicate entries keep the most recent with highest reviews  and delete the old entries 

In [6]:
# Below functions Count the no of duplcate apps in a data set based on the index
def duplicate_apps(data,col_idx):
    unique_apps = []
    dup_apps = []
    for rows in data:
        if rows[col_idx] in unique_apps:
            dup_apps.append(rows[col_idx])
        else:
            unique_apps.append(rows[col_idx])
            
    return unique_apps,dup_apps


In [7]:
# Process android data 
android_unique_apps ,android_dups_apps = duplicate_apps(android_data,0)
print(android_head)
print("Unique apps in Android data are ", len(android_unique_apps))
print("Duplicate apps in Android data are ", len(android_dups_apps))

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
Unique apps in Android data are  9659
Duplicate apps in Android data are  1181


In [8]:
#Process ios data 
print(ios_head)
ios_unique_apps, ios_dups_apps = duplicate_apps(ios_data,1)
print("Unique apps in ios data are ", len(ios_unique_apps))
print("Duplicate apps in ios data are ", len(ios_dups_apps))
print(ios_dups_apps)

# for d in ios_data:
#     if d[1] in ios_dups_apps:
#         print(d)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']
Unique apps in ios data are  7195
Duplicate apps in ios data are  2
['Mannequin Challenge', 'VR Roller Coaster']


In [9]:
   
# Below function will clean the duplicates rows and returns the clean and duplciate data using unique apps with maximum review data 
def clean_data(data, app_idx, reviews_idx):
    clean_data = []
    duplicate_data = []
    already_added = []
    
    # Find the max reviews for each app 
    app_max_review= {}
    for row in data:
        app= row[app_idx]
        reviews = float(row[reviews_idx])
        if app in app_max_review :
            if app_max_review[app] < reviews:
                app_max_review[app] = reviews
        else: 
            app_max_review[app]= reviews
            
    # Process the data based on max reviews as clean and duplicate data 
    for row in data:
        app= row[app_idx]
        reviews = float(row[reviews_idx])
        if app_max_review[app]==reviews and app not in already_added:
            clean_data.append(row)
            already_added.append(app)
        else:
            duplicate_data.append(row)
    return clean_data,duplicate_data

In [10]:
#Android clean data 
android_clean_data,android_dups_data =clean_data(android_data, 0,3)
print("length of android data is ", len(android_data))
print("Total no of unique rows in Android  data ",len(android_clean_data))
print("Total no of duplciate rows in Android  data ",len(android_dups_data))

length of android data is  10840
Total no of unique rows in Android  data  9659
Total no of duplciate rows in Android  data  1181


In [11]:
# ios Clean data 
ios_clean_data,ios_dups_data =clean_data(ios_data, 1,5)
print("length of ios data is ", len(ios_data))
print("Total no of unique rows in ios clean data ",len(ios_clean_data))
print("Total no of duplciate rows in ios  data ",len(ios_dups_data))

length of ios data is  7197
Total no of unique rows in ios clean data  7195
Total no of duplciate rows in ios  data  2


# Remove Non-English apps


In [64]:
#The below function will find if there are more than 3  non english characters 
def isenglish(str):
    char_count = 0 
    for c in str:
        if not (0<ord(c)<128) :# if any character in the app name has ascii character not berween 0 and 127 (means non english)
            char_count+=1
    if char_count>3:
        return False
        
    return True

# Remove non english apps from Android data 
android_english_data = []
for row in android_clean_data:
    if isenglish(row[0]):
        android_english_data.append(row)
        
print("Total Android english apps are ", len(android_english_data))

# Remove non english apps from ios data 
ios_english_data = []
for row in ios_clean_data:
    if isenglish(row[1]):
        ios_english_data.append(row)
        
        
print("Total ios english apps are ", len(ios_english_data))


Total Android english apps are  9614
Total ios english apps are  6181


# Isolating the Free Apps

As we mentioned in the introduction, we only build apps that are free to download and install, and our main source of revenue consists of in-app ads. Our data sets contain both free and non-free apps, and we'll need to isolate only the free apps for our analysis. Below, we isolate the free apps for both our data sets. 

So far we have cleaned the data by removing inacurate data , duplicate entries and non English apps. Now the next step in cleaning is to identify only free apps
 

In [13]:
android_free_apps_data  = []
ios_free_apps_data  = []
        
for row in android_english_data:
    if row[7] == '0':
        android_free_apps_data.append(row)
        
        
for row in ios_english_data:
    if row[4] == '0.0':
        ios_free_apps_data.append(row)
        
        
print("Free apps in android data using price columns ", len(android_free_apps_data))
print("Free apps in ios data using Price columns ", len(ios_free_apps_data))


Free apps in android data using price columns  8864
Free apps in ios data using Price columns  3220


# Most Common Apps by Genre

## Part One

As we mentioned in the introduction, our aim is to determine the kinds of apps that are likely to attract more users because our revenue is highly influenced by the number of people using our apps.

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:

   1. Build a minimal Android version of the app, and add it to Google Play.
   2. If the app has a good response from users, we then develop it further. 
   3. If the app is profitable after six months, we also build an iOS version of the app and add it to the App Store.

Because our end goal is to add the app on both the App Store and Google Play, we need to find app profiles that are successful on both markets. For instance, a profile that might work well for both markets might be a productivity app that makes use of gamification.

Let's begin the analysis by getting a sense of the most common genres for each market. For this, we'll build a frequency table for the prime_genre column of the App Store data set, and the Genres and Category columns of the Google Play data set.

In [14]:
# below function returns frequency of apps by genre 
def apps_freq_genre(data,index):
    apps_freq = {}
    
    for row in data:
        
        genre = row[index]
        if genre in apps_freq:
            apps_freq[genre]+=1
        else:
            apps_freq[genre]= 1
    
    apps_freq_percent = {}
    for app in apps_freq:
        apps_freq_percent[app] = (apps_freq[app]/len(data))*100
        
    return apps_freq_percent

def print_freq_data(data,index):
    freq_table = apps_freq_genre(data,index)
    freq_table= sorted(freq_table.items(), key= lambda kv:( kv[1],kv[0]),reverse= True)
    
    for app in freq_table:
        print("{} : {}".format( app[0], app[1]))
    




In [15]:
print(ios_head)
print("\nMost frequent Generes  in ios based on Prime_genere data  ")
print_freq_data(ios_free_apps_data,-5 )

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']

Most frequent Generes  in ios based on Prime_genere data  
Games : 58.13664596273293
Entertainment : 7.888198757763975
Photo & Video : 4.968944099378882
Education : 3.6645962732919255
Social Networking : 3.291925465838509
Shopping : 2.608695652173913
Utilities : 2.515527950310559
Sports : 2.142857142857143
Music : 2.049689440993789
Health & Fitness : 2.018633540372671
Productivity : 1.7391304347826086
Lifestyle : 1.5838509316770186
News : 1.3354037267080745
Travel : 1.2422360248447204
Finance : 1.1180124223602486
Weather : 0.8695652173913043
Food & Drink : 0.8074534161490683
Reference : 0.5590062111801243
Business : 0.5279503105590062
Book : 0.43478260869565216
Navigation : 0.18633540372670807
Medical : 0.18633540372670807
Catalogs : 0.12422360248447205


The most frequent apps in ios are **Games**

In [16]:
print(android_head)
print("\nMost frequent Category in android using category column ")
print_freq_data(android_free_apps_data,1)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']

Most frequent Category in android using category column 
FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_

The most frequent apps in Android  are **Family**

In [17]:
print(android_head)
print("\nMost frequent Category in android using generes column ")
print_freq_data(android_free_apps_data,-4)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']

Most frequent Category in android using generes column 
Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.39

The frequency tables we analyzed on the previous screen showed us that the App Store is dominated by apps designed for **fun**, while Google Play shows a more balanced landscape of both **practical and fun apps**. Now, we'd like to get an idea about the kind of apps with the most users.

In [18]:
print("\n ---android---")
print(android_head)
print(android_free_apps_data[:2])
print("\n ---ios---")
print(ios_head)
print(ios_free_apps_data[:2])


 ---android---
['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
[['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up'], ['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']]

 ---ios---
['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']
[['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1'], ['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558',

# Most Popular Apps by Genre on the App Store

One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the Installs column, but for the App Store data set this information is missing. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the rating_count_tot app.

Below, we calculate the average number of user ratings per app genre on the App Store:



In [19]:
# Build a freq table based on the apps isntalls and genres 
genres_ios =  apps_freq_genre(ios_english_data,-5)
genres_rating = {}
for row in ios_free_apps_data:
    genre = row[-5]
    rating = float(row[5])
    if genre in genres_rating:
        genres_rating[genre][0]+=1
        genres_rating[genre][1]+=rating
        
    else:
        genres_rating[genre]= [1,rating]
        
genres_avg_rating = {}        
for genre in genres_rating:
    genres_avg_rating[genre]= genres_rating[genre][1]/genres_rating[genre][0]
    
    
genres_avg_rating = sorted(genres_avg_rating.items(),key= lambda kv:(kv[1],kv[0]),reverse=True )   
for gen_rat in genres_avg_rating:
    print(gen_rat[0],":", gen_rat[1])
        


Navigation : 86090.33333333333
Reference : 74942.11111111111
Social Networking : 71548.34905660378
Music : 57326.530303030304
Weather : 52279.892857142855
Book : 39758.5
Food & Drink : 33333.92307692308
Finance : 31467.944444444445
Photo & Video : 28441.54375
Travel : 28243.8
Shopping : 26919.690476190477
Health & Fitness : 23298.015384615384
Sports : 23008.898550724636
Games : 22812.92467948718
News : 21248.023255813954
Productivity : 21028.410714285714
Utilities : 18684.456790123455
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Business : 7491.117647058823
Education : 7003.983050847458
Catalogs : 4004.0
Medical : 612.0


On average, navigation apps have the highest number of user reviews, but this figure is heavily influenced by Waze and Google Maps, which have close to half a million user reviews together:

In [20]:
# Apps from navigation and their ratings 
for app in ios_free_apps_data:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5]) # print name and number of ratings


Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


The same pattern applies to social networking apps, where the average number is heavily influenced by a few giants like Facebook, Pinterest, Skype, etc. Same applies to music apps, where a few big players like Pandora, Spotify, and Shazam heavily influence the average number.

Our aim is to find popular genres, but navigation, social networking or music apps might seem more popular than they really are. The average number of ratings seem to be skewed by very few apps which have hundreds of thousands of user ratings, while the other apps may struggle to get past the 10,000 threshold. We could get a better picture by removing these extremely popular apps for each genre and then rework the averages, but we'll leave this level of detail for later.

In [21]:
# Apps from Reference and their ratings 
for app in ios_free_apps_data:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5]) # print name and number of ratings


Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


In [22]:
# Apps from Social networking 
for app in ios_free_apps_data:
    if app[-5] == 'Social Networking':
        print(app[1], ':', app[5]) # print name and number of ratings


Facebook : 2974676
Pinterest : 1061624
Skype for iPhone : 373519
Messenger : 351466
Tumblr : 334293
WhatsApp Messenger : 287589
Kik : 260965
ooVoo – Free Video Call, Text and Voice : 177501
TextNow - Unlimited Text + Calls : 164963
Viber Messenger – Text & Call : 164249
Followers - Social Analytics For Instagram : 112778
MeetMe - Chat and Meet New People : 97072
We Heart It - Fashion, wallpapers, quotes, tattoos : 90414
InsTrack for Instagram - Analytics Plus More : 85535
Tango - Free Video Call, Voice and Chat : 75412
LinkedIn : 71856
Match™ - #1 Dating App. : 60659
Skype for iPad : 60163
POF - Best Dating App for Conversations : 52642
Timehop : 49510
Find My Family, Friends & iPhone - Life360 Locator : 43877
Whisper - Share, Express, Meet : 39819
Hangouts : 36404
LINE PLAY - Your Avatar World : 34677
WeChat : 34584
Badoo - Meet New People, Chat, Socialize. : 34428
Followers + for Instagram - Follower Analytics : 28633
GroupMe : 28260
Marco Polo Video Walkie Talkie : 27662
Miitomo : 2

However, this niche seems to show some potential. One thing we could do is take another popular book and turn it into an app where we could add different features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes about the book, etc. On top of that, we could also embed a dictionary within the app, so users don't need to exit our app to look up words in an external app.

This idea seems to fit well with the fact that the App Store is dominated by for-fun apps. This suggests the market might be a bit saturated with for-fun apps, which means a practical app might have more of a chance to stand out among the huge number of apps on the App Store.

Other genres that seem popular include weather, book, food and drink, or finance. The book genre seem to overlap a bit with the app idea we described above, but the other genres don't seem too interesting to us:

    Weather apps — people generally don't spend too much time in-app, and the chances of making profit from in-app adds are low. Also, getting reliable live weather data may require us to connect our apps to non-free APIs.

    Food and drink — examples here include Starbucks, Dunkin' Donuts, McDonald's, etc. So making a popular food and drink app requires actual cooking and a delivery service, which is outside the scope of our company.

    Finance apps — these apps involve banking, paying bills, money transfer, etc. Building a finance app requires domain knowledge, and we don't want to hire a finance expert just to build an app.

Now let's analyze the Google Play market a bit.

# Most Popular Apps by Genre on Google Play

For the Google Play market, we actually have data about the number of installs, so we should be able to get a clearer picture about genre popularity. However, the install numbers don't seem precise enough — we can see that most values are open-ended (100+, 1,000+, 5,000+, etc.):

In [23]:
print_freq_data(android_free_apps_data,5) #Install column 

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


In [57]:
print(android_head,"\n\n")
#print(android_data[:5])
# genrate  a freq table with category wise  average installs

category_android =  apps_freq_genre(android_free_apps_data,5)
print(category_android)
category_installs = {}
for row in android_free_apps_data:
    category = row[1]
    installs = row[5]
    installs = installs.replace(',','')
    installs = installs.replace('+','')
    installs = float(installs)
                     
    if category in category_installs:
        category_installs[category][0]+=1
        category_installs[category][1]+=installs
        
    else:
        category_installs[category]= [1,installs]
        
category_avg_installs = {}        
for category in category_installs:
    category_avg_installs[category]= category_installs[category][1]/category_installs[category][0]
    
#print(genres_installs)  
category_avg_installs = sorted(category_avg_installs.items(),key= lambda kv:(kv[1],kv[0]),reverse=True )   
for cat_rat in category_avg_installs:
    print(cat_rat[0],":", cat_rat[1])
        


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver'] 


{'10,000+': 10.198555956678701, '5,000,000+': 6.825361010830325, '50,000,000+': 2.3014440433213, '100,000+': 11.552346570397113, '50,000+': 4.7721119133574, '1,000,000+': 15.726534296028879, '10,000,000+': 10.548285198555957, '5,000+': 4.512635379061372, '500,000+': 5.561823104693141, '1,000,000,000+': 0.22563176895306858, '100,000,000+': 2.1322202166064983, '1,000+': 8.393501805054152, '500,000,000+': 0.2707581227436823, '500+': 3.2490974729241873, '100+': 6.915613718411552, '50+': 1.917870036101083, '10+': 3.5424187725631766, '1+': 0.5076714801444043, '5+': 0.78971119133574, '0+': 0.04512635379061372, '0': 0.01128158844765343}
COMMUNICATION : 38456119.167247385
VIDEO_PLAYERS : 24727872.452830188
SOCIAL : 23253652.127118643
PHOTOGRAPHY : 17840110.40229885
PRODUCTIVITY : 16787331.344927534
GAME : 15588015.603248259
TRAVEL_AND_LOCAL :

On average, communication apps have the most installs: 38,456,119. This number is heavily skewed up by a few apps that have over one billion installs (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts), and a few others with over 100 and 500 million installs:

In [40]:
for app in android_free_apps_data:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

In [42]:
#VIDEO_PLAYERS
for app in android_free_apps_data:
    if app[1] == 'VIDEO_PLAYERS' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

YouTube : 1,000,000,000+
Motorola Gallery : 100,000,000+
VLC for Android : 100,000,000+
Google Play Movies & TV : 1,000,000,000+
MX Player : 500,000,000+
Dubsmash : 100,000,000+
VivaVideo - Video Editor & Photo Movie : 100,000,000+
VideoShow-Video Editor, Video Maker, Beauty Camera : 100,000,000+
Motorola FM Radio : 100,000,000+


In [44]:
#Social 
for app in android_free_apps_data:
    if app[1] == 'SOCIAL' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Facebook : 1,000,000,000+
Facebook Lite : 500,000,000+
Tumblr : 100,000,000+
Pinterest : 100,000,000+
Google+ : 1,000,000,000+
Badoo - Free Chat & Dating App : 100,000,000+
Tango - Live Video Broadcast : 100,000,000+
Instagram : 1,000,000,000+
Snapchat : 500,000,000+
LinkedIn : 100,000,000+
Tik Tok - including musical.ly : 100,000,000+
BIGO LIVE - Live Stream : 100,000,000+
VK : 100,000,000+




We see the same pattern for the video players category, which is the runner-up with 24,727,872 installs. The market is dominated by apps like Youtube, Google Play Movies & TV, or MX Player. The pattern is repeated for social apps (where we have giants like Facebook, Instagram, Google+, etc.), photography apps (Google Photos and other popular photo editors), or productivity apps (Microsoft Word, Dropbox, Google Calendar, Evernote, etc.).

Again, the main concern is that these app genres might seem more popular than they really are. Moreover, these niches seem to be dominated by a few giants who are hard to compete against.

The game genre seems pretty popular, but previously we found out this part of the market seems a bit saturated, so we'd like to come up with a different app recommendation if possible.

The books and reference genre looks fairly popular as well, with an average number of installs of 8,767,811. It's interesting to explore this in more depth, since we found this genre has some potential to work well on the App Store, and our aim is to recommend an app genre that shows potential for being profitable on both the App Store and Google Play.

Let's take a look at some of the apps from this genre and their number of installs:


In [45]:
#Books and reference 
for app in android_free_apps_data:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+




However, it looks like there are only a few very popular apps, so this market still shows potential. Let's try to get some app ideas based on the kind of apps that are somewhere in the middle in terms of popularity (between 1,000,000 and 100,000,000 downloads):


In [47]:

for app in android_free_apps_data:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])



Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

In [60]:
print(android_head,"\n\n")
#print(android_data[:5])
# genrate  a freq table with category wise  average installs

genre_android =  apps_freq_genre(android_free_apps_data,-4)
genre_installs = {}
for row in android_free_apps_data:
    genre = row[-4]
    installs = row[5]
    installs = installs.replace(',','')
    installs = installs.replace('+','')
    installs = float(installs)
                     
    if genre in genre_installs:
        genre_installs[genre][0]+=1
        genre_installs[genre][1]+=installs
        
    else:
        genre_installs[genre]= [1,installs]
        
genre_avg_installs = {}        
for genre in genre_installs:
    genre_avg_installs[category]= genre_installs[genre][1]/genre_installs[genre][0]
    
#print(genres_installs)  
genre_avg_installs = sorted(genre_avg_installs.items(),key= lambda kv:(kv[1],kv[0]),reverse=True )   
for gen_rat in category_avg_installs:
    print(gen_rat[0],":", gen_rat[1])
        

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver'] 


COMMUNICATION : 38456119.167247385
VIDEO_PLAYERS : 24727872.452830188
SOCIAL : 23253652.127118643
PHOTOGRAPHY : 17840110.40229885
PRODUCTIVITY : 16787331.344927534
GAME : 15588015.603248259
TRAVEL_AND_LOCAL : 13984077.710144928
ENTERTAINMENT : 11640705.88235294
TOOLS : 10801391.298666667
NEWS_AND_MAGAZINES : 9549178.467741935
BOOKS_AND_REFERENCE : 8767811.894736841
SHOPPING : 7036877.311557789
PERSONALIZATION : 5201482.6122448975
WEATHER : 5074486.197183099
HEALTH_AND_FITNESS : 4188821.9853479853
MAPS_AND_NAVIGATION : 4056941.7741935486
FAMILY : 3695641.8198090694
SPORTS : 3638640.1428571427
ART_AND_DESIGN : 1986335.0877192982
FOOD_AND_DRINK : 1924897.7363636363
EDUCATION : 1833495.145631068
BUSINESS : 1712290.1474201474
LIFESTYLE : 1437816.2687861272
FINANCE : 1387692.475609756
HOUSE_AND_HOME : 1331540.5616438356
DATING : 854028.830



This niche seems to be dominated by software for processing and reading ebooks, as well as various collections of libraries and dictionaries, so it's probably not a good idea to build similar apps since there'll be some significant competition.

We also notice there are quite a few apps built around the book Quran, which suggests that building an app around a popular book can be profitable. It seems that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets.

However, it looks like the market is already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.
#  Conclusions

In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets.

We concluded that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets. The markets are already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.
